src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java

```diff
@@ -113,6 +114,8 @@ public final class ProxyMain {
 
     private static final int STATS_LOG_REPEAT_INTERVAL = 60;
 
+    private static AddOnStatusDiagnostics addOnStatus;
```

Comment:
@jansu76 it seems that this is almost god class. Were there any discussions about making it more granular?

Reply:
I am not aware of any discussions. Refactoring might do it good, e.g. moving the different handlers to separate classes. As far as god classes go though, I'm afraid this isn't even close to the level of worst ones...

The changes from this PR are quite small so I think I would outscope refactoring to a separate ticket (or if you are real keen on doing those now, I can let you do the modifications).

Reply:
Understood, for sure it's out of scope, but was curious.